src/org/jetbrains/plugins/scala/lang/refactoring/introduceVariable/ScopeSuggester.scala

```diff
@@ -112,51 +129,94 @@ object ScopeSuggester {
+    }
 
-      val resultBuffer = new ArrayBuffer[ArrayBuffer[PsiFile]]()
-      words.foreach(oneRound(_, resultBuffer))
+    result.toArray
+  }
 
-      var intersectionResult = resultBuffer(0)
-      def intersect(inBuffer: ArrayBuffer[PsiFile]) = {
-        intersectionResult = intersectionResult.intersect(inBuffer)
-      }
+  def handleOnePackage(typeElement: ScTypeElement, inPackageName: String, containinDirectory: PsiDirectory,
+                       conflictsReporter: ConflictsReporter, project: Project, editor: Editor, isReplaceAll: Boolean, inputName: String): PackageScopeItem = {
+    def getFilesToSearchIn(currentDirectory: PsiDirectory): Array[PsiFile] = {
+      if (!isReplaceAll) {
+        Array(typeElement.getContainingFile)
+      } else {
+        def oneRound(word: String, bufResult: ArrayBuffer[ArrayBuffer[PsiFile]]) = {
+          val buffer = new ArrayBuffer[PsiFile]()
+
+          val processor = new Processor[PsiFile] {
+            override def process(file: PsiFile): Boolean = {
+              buffer += file
+              true
+            }
+          }
+
+          val helper: PsiSearchHelper = PsiSearchHelper.SERVICE.getInstance(typeElement.getProject)
+          helper.processAllFilesWithWord(word, GlobalSearchScopesCore.directoryScope(currentDirectory, true), processor, true)
+
+          bufResult += buffer
+        }
+
+        val typeName = typeElement.calcType.presentableText
+        var words = Array[String]()
+        words = StringUtil.getWordsIn(typeName).toArray(words)
 
-      resultBuffer.foreach((element: ArrayBuffer[PsiFile]) => intersect(element))
-      intersectionResult.toList.reverse.toArray
+        val resultBuffer = new ArrayBuffer[ArrayBuffer[PsiFile]]()
+        words.foreach(oneRound(_, resultBuffer))
+
+        var intersectionResult = resultBuffer(0)
+        def intersect(inBuffer: ArrayBuffer[PsiFile]) = {
+          intersectionResult = intersectionResult.intersect(inBuffer)
+        }
+
+        resultBuffer.foreach((element: ArrayBuffer[PsiFile]) => intersect(element))
+        intersectionResult.toList.reverse.toArray
+      }
     }
 
+    val inPackage = ScPackageImpl.findPackage(typeElement.getProject, inPackageName)
     val projectSearchScope = GlobalSearchScope.projectScope(typeElement.getProject)
-    val packageReal = ScPackageImpl.findPackage(typeElement.getProject, packageName)
-
-    val packageObject = packageReal.findPackageObject(projectSearchScope)
+    val packageObject = inPackage.findPackageObject(projectSearchScope)
 
     val fileEncloser = if (packageObject.isDefined)
       PsiTreeUtil.getChildOfType(PsiTreeUtil.getChildOfType(packageObject.get, classOf[ScExtendsBlock]), classOf[ScTemplateBody])
     else
-      null
+      containinDirectory
 
     val allOcurrences: mutable.MutableList[Array[ScTypeElement]] = mutable.MutableList()
     val allValidators: mutable.MutableList[ScalaTypeValidator] = mutable.MutableList()
```

Comment:
In the next method you actually work only with scala files. It's better to check it first and cast it in the new variable.